layers/eight_mile/tf/layers.py

```diff
@@ -848,6 +1094,8 @@ def output_fn(self, rnnout, state):
     def call(self, inputs):
         inputs, lengths = tensor_and_lengths(inputs)
         mask = tf.sequence_mask(lengths)
+        max_length = tf.reduce_max(lengths)
```

Comment:
I know this is a pretty easy thing to do (only two lines) but it seems like it will be sprinkled throughout a lot of the code, might make sense as a function?

Reply:
maybe a private one, op is different if (Bi)LSTM*1 or 2

Reply:
this is a good point, but I think it can be cleaned up later